app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -0,0 +1,51 @@
+module ForemanDeployments
+  module Tasks
+    class SearchTaskDefinition < BaseDefinition
+      class Action < BaseAction
+        def run
+          results = SearchTaskDefinition.create_object(input)
+          results = SerializableArray.new(results)
+          SearchTaskDefinition.create_output(results, output)
+        end
+      end
+
+      class SerializableArray < Array
+        def to_hash
+          {}
+        end
+      end
+
+      def validate
+        results = SearchTaskDefinition.create_object(parameters)
+        messages = nil
+        unless results.any?
+          messages = ["#{parameters[:klass]}.search_for(\"#{parameters[:search_term]}\") didnt return valid objects"]
```

Comment:
The same problem here. Please see my comment here https://github.com/theforeman/foreman_deployments/pull/9/files#r38302101